File: ruby/hackerrank/minimum_distances.rb
#!/bin/ruby

n = gets.strip.to_i
A = gets.strip.split(' ').map(&:to_i)
orig_min = 900000000
min = orig_min
A.each_with_index do |i, ii|
    A.each_with_index do |j, jj|
        if i == j && ii != jj && (v = (ii - jj).abs) < min
            min = v
        end
    end
end

if min == orig_min
    print -1
else
    print min
end

File: javascript/hackerrank/minimum_distances.js
process.stdin.resume();
process.stdin.setEncoding('ascii');

var input_stdin = "";
var input_stdin_array = "";
var input_currentline = 0;

process.stdin.on('data', function (data) {
    input_stdin += data;
});

process.stdin.on('end', function () {
    input_stdin_array = input_stdin.split("\n");
    main();
});

function readLine() {
    return input_stdin_array[input_currentline++];
}

/////////////// ignore above this line ////////////////////

function main() {
    var n = parseInt(readLine());
    A = readLine().split(' ');
    A = A.map(Number);
    min = -1;
    for(i = 0; i < A.length; i++) {
        for(j=0; j < A.length; j++) {
            v = Math.abs(i - j);
            if (i != j && A[i] == A[j] && (v < min || min == -1)) {
                min = v;
            }
        }
    }
    console.log(min);
}
